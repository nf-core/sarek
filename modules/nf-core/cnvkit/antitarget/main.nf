process CNVKIT_ANTITARGET {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/cnvkit:0.9.11--pyhdfd78af_0':
        'biocontainers/cnvkit:0.9.11--pyhdfd78af_0' }"

    input:
    tuple val(meta), path(targets)

    output:
    tuple val(meta), path("*.bed"), emit: bed
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    cnvkit.py \\
        antitarget \\
        $targets \\
        --output ${prefix}.antitarget.bed \\
        $args

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        cnvkit: \$(cnvkit.py version | sed -e "s/cnvkit v//g")
    END_VERSIONS
    """
}
