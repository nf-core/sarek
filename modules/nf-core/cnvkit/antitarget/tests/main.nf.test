nextflow_process {

    name "Test Process CNVKIT_ANTITARGET"
    script "../main.nf"
    process "CNVKIT_ANTITARGET"

    tag "modules"
    tag "modules_nfcore"
    tag "cnvkit"
    tag "cnvkit/antitarget"

    test("human - bed") {

        when {
            process {
                """
                input[0] = [
                    [id: 'test'],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/multi_intervals.bed', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("human - bed - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [id: 'test'],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/multi_intervals.bed', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
