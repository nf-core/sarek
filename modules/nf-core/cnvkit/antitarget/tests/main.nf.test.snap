{
    "human - bed - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.antitarget.bed:md5,3d4d20f9f23b39970865d29ef239d20b"
                    ]
                ],
                "1": [
                    "versions.yml:md5,b5e73ea85743cedc68ca6ef8006e5030"
                ],
                "bed": [
                    [
                        {
                            "id": "test"
                        },
                        "test.antitarget.bed:md5,3d4d20f9f23b39970865d29ef239d20b"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b5e73ea85743cedc68ca6ef8006e5030"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-05-13T13:58:45.758206"
    },
    "human - bed": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.antitarget.bed:md5,3d4d20f9f23b39970865d29ef239d20b"
                    ]
                ],
                "1": [
                    "versions.yml:md5,b5e73ea85743cedc68ca6ef8006e5030"
                ],
                "bed": [
                    [
                        {
                            "id": "test"
                        },
                        "test.antitarget.bed:md5,3d4d20f9f23b39970865d29ef239d20b"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b5e73ea85743cedc68ca6ef8006e5030"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-06-10T10:32:33.936192"
    }
}